src/components/SelectVirtualized/index.tsx

```diff
@@ -0,0 +1,329 @@
+  getGroupLabel,
+  getOptionLabel: externalGetOptionLabel,
+  getOptionSelected,
+  groupBy,
+  label,
+  loading = false,
+  multiple = false,
+  onChange: externalOnChange,
+  onClose: externalOnClose,
+  onInputChange: externalOnInputChange,
+  onScrollEnd,
+  options: externalOptions,
+  placeholder,
+  popperWidth,
+  required = false,
+  size = "medium",
+  style,
+  type = "text",
+  value = null,
+  variant = "outlined",
+  optionsNumber,
+}: ISelect<T>) => {
+  const getOptionLabel = useCallback(
+    (option: T): string => (externalGetOptionLabel ? externalGetOptionLabel(option) : `${option}`),
+    [externalGetOptionLabel]
+  );
+
+  const isOptionSelected = useCallback(
+    (option: T, value: T): boolean => {
+      if (getOptionSelected) {
+        return getOptionSelected(option, value);
+      }
+
+      return !!value && option === value;
+    },
+    [getOptionSelected]
+  );
+
+  const onChange = useCallback(
+    // TODO#lb: fix any type
+    (event: any, value: any) => {
+      suppressEvent(event);
+      externalOnChange(value);
+      console.log(value);
+    },
+    [externalOnChange]
+  );
+
+  const onClose = useCallback(
+    // TODO#lb: fix any type
+    (event: any) => {
+      if (!externalOnClose) {
+        return;
+      }
+
+      suppressEvent(event);
+      externalOnClose();
+    },
+    [externalOnClose]
+  );
+
+  const onInputChange = useCallback(
+    // TODO#lb: fix any type
+    (event: any, value: string) => {
+      if (!externalOnInputChange) {
+        return;
+      }
+
+      suppressEvent(event);
+      externalOnInputChange(value);
+    },
+    [externalOnInputChange]
+  );
+
+  /*const onScroll = useCallback(
+    ({ currentTarget: listboxNode }: SyntheticEvent) => {
+      if (!listboxNode || !onScrollEnd) {
+        return;
+      }
+
+      const { clientHeight, scrollHeight, scrollTop } = listboxNode;
+      const scrollEnded = clientHeight + scrollTop === scrollHeight;
+      if (scrollEnded) {
+        onScrollEnd();
+      }
+    },
+    [onScrollEnd]
+  );*/
+
+  const ROW = ({ data, index, style }: any) => {
+    return <div style={style}>{data[index]}</div>;
+  };
+
+  function useResetCache(data: any) {
+    const ref = React.useRef<VariableSizeList>(null);
+    React.useEffect(() => {
+      if (ref.current != null) {
+        ref.current.resetAfterIndex(0, true);
+      }
+    }, [data]);
+    return ref;
+  }
+
+  // Adapter for react-window
+  const listboxComponent = React.forwardRef<HTMLDivElement, React.HTMLAttributes<HTMLElement>>(
+    function ListboxComponent(props, ref) {
+      const { children, ...other } = props;
+      const itemData: React.ReactChild[] = [];
+      (children as React.ReactChild[]).forEach((item: React.ReactChild & { children?: React.ReactChild[] }) => {
+        itemData.push(item);
+        itemData.push(...(item.children || []));
+      });
+
+      const gridRef = useResetCache(options.length);
+
+      const adaptedHeight = useMemo(() => {
```

Comment:
Here you can use the result of `adaptedSize` for the calculation, code will simplify